### PR TITLE
fixup some dependencies

### PR DESCRIPTION
**Pull request by DinoChiesa**
- move es6-template-strings from devDependencies to dependencies in package.json. It is required by users of this module at runtime, not only for developers of this module.

- add an engines field in package.json, and specify npm 8.3.0 because we will use the overrides feature.

- use an overrides field in package.json to override some vulnerable and outdated modules.

- update required mocha to 10.0.0 to avoid some vulnerable dependencies of earlier versions of mocha.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,16 +49,16 @@
     },
     "homepage": "https://github.com/ianpogi5/mocha-badge-generator",
     "dependencies": {
-        "badge-up": "^3.0.0",
+        "badge-up": "https://registry.npmjs.org/@rpl/badge-up/-/badge-up-3.0.0.tgz",
```
Comment (brettz9) [inline package.json]: I think this should work:

```suggestion
        "badge-up": "npm:@rpl/badge-up@3.0.0",
```

File: package.json [hunk 2]
```diff
@@ -71,5 +71,14 @@
         "trailingComma": "all",
         "bracketSpacing": true,
         "arrowParens": "avoid"
+    },
+    "overrides": {
+        "nth-check": "^2.0.1",
+        "package-json": "^8.1.0",
+        "uuid": "^7.0.3"
+    },
+    "engines": {
```
Comment (brettz9) [inline package.json]: There is already an `engines` property, so if you could add these there instead...

**Review by brettz9 — CHANGES_REQUESTED** [review]
Looks good, thank you. If you could just adapt per the comments...

**Comment by brettz9** [comment]
@ianpogi5 : Could you publish a new version? npm now requires 2FA, and although I set it up for myself, apparently successfully, it says I don't have permission after correctly entering the One-Time Password (OTP).

**Comment by brettz9** [comment]
@ianpogi5 : Do you think you could take a look?

**Comment by brettz9** [comment]
@ianpogi5 : After the merge, I guess it will be ready for the stable 0.10.3 release? (or 0.10.6 as you have bumped it there)

**Comment by ianpogi5** [comment]
I'll do a release for you this time but I'll take a screenshot on what I did. 

Yes you should study GA. I make your life easier.

**Pull request merged**